pkg/service/gcp/kubec.go

```diff
@@ -0,0 +1,156 @@
+package gcp
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+
+	"github.com/pkg/errors"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+	"google.golang.org/genproto/googleapis/container/v1"
+	"k8s.io/client-go/tools/clientcmd"
+
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+func (g *GCPController) getToken(ctx context.Context, req *proto.GetTokenRequest) (*proto.GetTokenResponse, error) {
+
+	cred, err := getCredentials(ctx, req.AccountName)
+	if err != nil {
+		return nil, errors.Wrap(err, "getToken")
+	}
+	cluster, err := g.getClusterInternal(ctx, cred, req.Region, req.ClusterName)
+
+	if err != nil {
+		return nil, errors.Wrap(err, "GetToken")
+	}
+
+	if cluster.Status != container.Cluster_RUNNING {
+		return nil, fmt.Errorf("cluster is not in running state yet")
+	}
+	server := fmt.Sprintf("https://%s", cluster.Endpoint)
+
+	g.logger.Info(ctx, "cluster config", "server", server)
+
+	scopes := []string{}
+	nodePools := cluster.NodePools
+
+	if len(nodePools) > 0 {
+		scopes = append(scopes, nodePools[0].Config.OauthScopes...)
+	}
+
+	auth, err := getAuthClient(ctx, cred, scopes)
+	if err != nil {
+		g.logger.Error(ctx, "failed to get auth2 client", "error", err)
+		return nil, errors.Wrap(err, "GetToken")
+
+	}
+
+	t, err := auth.TokenSource.Token()
+	if err != nil {
+		g.logger.Error(ctx, "failed to get token from token source", "error", err)
+		return nil, errors.Wrap(err, "GetToken")
+	}
+	token := t.AccessToken
+	return &proto.GetTokenResponse{
+		Token:    token,
+		Endpoint: server,
+		CaData:   cluster.MasterAuth.GetClusterCaCertificate(),
+	}, nil
+}
+
+func (g *GCPController) getKubeConfig(ctx context.Context, req *proto.GetKubeConfigRequest) (*proto.GetKubeConfigResponse, error) {
+
+	cred, err := getCredentials(ctx, req.AccountName)
+	if err != nil {
+		return nil, errors.Wrap(err, "getToken:")
+	}
+	cluster, err := g.getClusterInternal(ctx, cred, req.Region, req.ClusterName)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if cluster.Status != container.Cluster_RUNNING {
+		return nil, fmt.Errorf("cluster is not in running state yet")
+	}
+	server := fmt.Sprintf("https://%s", cluster.Endpoint)
+
+	g.logger.Info(ctx, "cluster config", "server", server)
+
+	scopes := []string{}
+	nodePools := cluster.NodePools
+
+	if len(nodePools) > 0 {
+		scopes = append(scopes, nodePools[0].Config.OauthScopes...)
+	}
+
+	auth, err := getAuthClient(ctx, cred, scopes)
+	if err != nil {
+		g.logger.Error(ctx, "failed to get auth2 client", "error", err)
+		return nil, errors.Wrap(err, "GetKubeConfig")
+	}
+
+	t, err := auth.TokenSource.Token()
+	if err != nil {
+		g.logger.Error(ctx, "failed to get token from token source", "error", err)
+		return nil, errors.Wrap(err, "GetKubeConfig")
+
+	}
```

Comment:
Can we call `getToken` and use the values it returns here?

Reply:
yup, leaving it for now. Can refactor for better. Ill keep this as a open issue.

Reply:
Not a big refactor, can we do it now? Else will lose track.